articles/tutorials/building_2d_games/23_completing_the_game/index.md

```diff
@@ -0,0 +1,428 @@
+
+### GameSceneUI Fields
+
+Add the following fields to the `GameSceneUI` class:
+
+[!code-csharp[](./snippets/gamesceneui/fields.cs)]
+
+Let's break down what each of these fields is responsible for:
+
+- `s_scoreFormat`: A string format template used to display the player's score with leading zeros.
+- `_uiSoundEffect`: Stores the sound effect played for UI interactions like button clicks and focus changes.
+- `_overlay`: A semi-transparent overlay that darkens the game screen when a menu is displayed.
+- `_pausePanel`: The panel containing the UI elements shown when the game is paused.
+- `_resumeButton`: A reference to the resume button, allowing us to set focus on it when the pause panel is shown.
+- `_gameOverPanel`: The panel containing the UI elements shown when a game over occurs.
+- `_retryButton`: A reference to the retry button, allowing us to set focus to it when the game over panel is shown.
+- `_scoreText`: The text display showing the player's current score.
+
+### GameSceneUI Events
+
+After the fields, add the following events to the `GameSceneUI` class:
+
+[!code-csharp[](./snippets/gamesceneui/events.cs)]
+
+These events allow the `GameSceneUI` class to notify the `GameScene` when important UI actions occur:
+
+- `ResumeButtonClick`: Triggered when the player clicks the Resume button on the pause panel.
+- `QuitButtonClick`: Triggered when the player clicks the Quit button on either panel.
+- `RetryButtonClick`: Triggered when the player clicks the Retry button on the game over panel.
+
+### GameSceneUI Constructor
+
+Add the following constructor to the `GameSceneUI` class after the events:
+
+[!code-csharp[](./snippets/gamesceneui/constructor.cs)]
+
+This constructor initializes all UI components:
+
+1. Set the container to fill the entire screen.
+2. Adds itself to Gum's root element.
+3. Loads necessary assets (sound effect and texture atlas).
+4. Creates and adds child elements in the correct order.
+
+### GameSceneUI UI Creation Methods
+
+Next add the methods to create the various UI elements that are managed by the `GameSceneUI` class.  Add the following method after the constructor:
+
+[!code-csharp[](./snippets/gamesceneui/createscoretext.cs)]
+
+This method takes creates a `TextRuntime` element that we can use to display the player's score and returns it back.  After this method, add the following method:
+
+[!code-csharp[](./snippets/gamesceneui/createoverlay.cs)]
+
+This method creates a `ColoredRectangleRuntime` element that is semi-transparent and returns it back.  This will be used to dim the game when the pause and game over panels are shown.  After this method, add the following method:
+
+[!code-csharp[](./snippets/gamesceneui/createpausepanel.cs)]
+
+This method builds the `Panel` that is shown hen the game is paused, including the "Resume" and "Quit" buttons, then returns it back. Finally, after this method, add the following method:
```

Comment:
No chickens were harmed in the creation of this paragraph I assume?  "hen" (Then?)